lib/puma/server.rb

```diff
@@ -258,6 +258,7 @@ def handle_servers_lopez_mode
         STDERR.puts e.backtrace
       ensure
         begin
+          @notify.close rescue nil
```

Comment:
Makes sense, seems like an oversight. Is this tested?

Reply:
The test suite did not previously test for leaky fd's.  Right or wrong, I decided that doing a `GC.start` & `ObjectSpace.each_object(IO)` for most tests might set the CI time out too far.  So I just did it when the tests finished, and made it easy to check individual tests if needed.

When I first added it, the fd's were well over 100.

Summing up, there are no asserts for leaky fd's at present, just the check at the end.